src/frontend/src/pages/FinancePage/ReimbursementRequestForm/ReimbursementFormView.tsx

```diff
@@ -75,6 +75,10 @@ const ReimbursementRequestFormView: React.FC<ReimbursementRequestFormViewProps>
   const [datePickerOpen, setDatePickerOpen] = useState(false);
   const toast = useToast();
   const products = watch(`reimbursementProducts`);
+  const expenseTypeId = watch('expenseTypeId');
+  const selectedExpenseType = allExpenseTypes.find((expenseType) => expenseType.expenseTypeId === expenseTypeId);
+  const refundSources = selectedExpenseType?.allowedRefundSources || [];
```

Comment:
since we're doing this this way, you should swap where expense type and refund source are in the frontend form rn so someone can selected an expense type before they select a refund source